src/test/java/steps/StepAssertor.java

```diff
@@ -76,28 +77,41 @@ private static boolean testCategoryCheck(String json, String responseObject, Str
 
     private static boolean testApplicantEmployees(String json, String responseObject, String key, String expected) {
         if (key.contains("employer name")) {
-            String jsonPath = "$.categoryChecks..[?(@.calculationType == '" + responseObject + "')]." + KEY_MAP.get(key);
+            String nino = key.replace("employer name -", "").trim().toUpperCase();
+            String jsonPath = "$.categoryChecks..[?(@.calculationType == '" + responseObject + "')].individuals..[?(@.nino == '" + nino + "')].employers";
             JSONArray employersWrapper = read(json, jsonPath);
+
             List<String> employers = new ArrayList<>();
             Iterator it = ((JSONArray) employersWrapper.get(0)).iterator();
             while (it.hasNext()) {
                 employers.add(it.next().toString().toLowerCase());
             }
             List<String> requiredEmployers = Arrays.stream(expected.split(",")).map(String::trim).collect(Collectors.toList());
-            assertTrue(employers.size() == requiredEmployers.size(), String.format("%s %s: expected %d employers but actual was %d\nJSON: %s", responseObject, key, requiredEmployers.size(), employers.size(), json));
-            assertTrue(employers.containsAll(requiredEmployers), String.format("%s %s: expected employers %s but actual was %s\nJSON: %s", responseObject, key, requiredEmployers, employers, json));
+
+            assertTrue(employers.size() == requiredEmployers.size(),
+                String.format("%s %s: expected %d employers but actual was %d\nJSON: %s", responseObject, key, requiredEmployers.size(), employers.size(), json));
+            assertTrue(employers.containsAll(requiredEmployers),
+                String.format("%s %s: expected employers %s but actual was %s\nJSON: %s", responseObject, key, requiredEmployers, employers, json));
             return true;
         }
         return false;
     }
 
-    private static boolean testApplicants(String json, String responseObject, String key, String expected) {
+    private static boolean testCategoryCheckNinos(String json, String responseObject, String key, String expected) {
+        if (responseObject.startsWith("Category ") && key.contains("nino")) {
+            String jsonPath = "$.categoryChecks..[?(@.calculationType == '" + responseObject + "')].individuals..[?(@.nino == '" + expected + "')]";
+            JSONArray jsonData = read(json, jsonPath);
+            assertTrue(jsonData.size() > 0, String.format("%s %s: Unable to find category check nino %s\nJSON: %s", responseObject, key, expected, json));
+            return true;
+        }
+        return false;
+    }
+
```

Comment:
This is hard to follow.  In general this class is becoming a bit bloated and difficult to understand - e.g. what do those complicated json paths even mean?  This could do with refactoring at some point soon-ish.